rsearch/conductor.go

```diff
@@ -0,0 +1,43 @@
+package rsearch
+
+func manageResources(ns Event, terminators map[string]chan Done, config Config, out chan Event) {
+	uid := ns.Object.Metadata.Uid
+	if ns.Type == "ADDED" {
+		done := make(chan Done)
+		terminators[uid] = done
+		ns.Object.Produce(out, terminators[uid], config)
+	} else if ns.Type == "DELETED" {
+		close(terminators[uid])
+		delete(terminators, uid)
+	} else if ns.Type == "_CRASH" {
+		for uid, c := range terminators {
+			close(c)
+			delete(terminators, uid)
+		}
+	}
+}
+
+// Conductor manages a set of goroutines one per namespace.
+func Conductor(in <-chan Event, done <-chan Done, config Config) <-chan Event {
```

Comment:
For starters, you might add a comment that explains what's happening. Then this could also help to clear up 'done' ambiguities.